Fix silent error (bootloop) due to flash chip size & MFRID

CarlosGS opened the pull request:
Currently there is a check for specific MFRID which badly freezes when using memory chips from any different brand.

It causes a white screen at startup, with no information to the user, which would cause soft bricks for printers using other memory chips when upgrading to the official 4.4.0+ firmware.

This was present since two years ago, and the problem simply has appeared now that the flash chip is actually used (with the new & awesome offloading of resources into the flash memory).
The fix is to remove the MFRID check, because checking for DEVID is enough to determine that the correct chip is present. With this change everything works perfectly :tada: 

Fixes [ Bootloader not getting updated #2188 ](https://github.com/prusa3d/Prusa-Firmware-Buddy/issues/2188)

**Update:** Also fixes a similar problem with 16MB flash chips, see https://github.com/prusa3d/Prusa-Firmware-Buddy/pull/2203#issuecomment-1201824784

MinnesotaMean reviewed (commented):
This fixed my issue for 4.4.0 updating. Awesome work!

DRracer commented:
@CarlosGS do you see a binary at `.dependencies/clang-format-9.0.0-noext/clang-format`? Does it have the executable flag set?
Normally, the build environment is set up using `utils/bootstrap.py`, which shall be called also during normal build via `utils/build.py`

CarlosGS committed:
Prevent lock when using flash chips with different MFRID

CarlosGS committed:
Flag variable as unused (CI)

CarlosGS commented:
@DRracer Thank you, gave it a fresh try and it works! :)
Strict compilation was failing and had to add `__attribute__((unused))` to the variable.

CarlosGS committed:
Include 16MB & 32MB flash sizes

CarlosGS commented:
Found a similar problem with boards that have a 16MB flash chip, the firmware was halting as it expected either 8MB or 64MB (fixed DEVID and sector count). This was tricky to identify as it occurs so early at boot. Thanks JetSet in Discord for helping troubleshoot!

This last commit completes the support in the flash driver, allowing a normal boot in these cases. It would still be possible to require specific memory sizes in the future, but at least the firmware won't freeze silently, allowing to eventually notify these users.

CarlosGS committed:
Merge branch 'prusa3d:master' into patch-1

PeterDHabermehl reviewed (approved):
Trying to update my Mini I ran into the bootloop issue.
I did some research on the internet and found @CarlosGS contribution.
I reviewed the changes made, considered it plausible and built a firmware from the recent master branch plus this commit. 
Build process and flashing of the fw worked flawless, so I approve this commit.

PeterDHabermehl reviewed (approved):
This commit properly addresses issues with different flash chip sizes.
Reviewed the code and built a fw to test - passed successfully, no issues, was able to flash two printers that failed before.

PeterDHabermehl reviewed (approved):
srry, forgot to mark the file as reviewed before.
I can confirm that the fix is working, reviewed the code and found it straightforward. No objections.

bmangodt commented:
This NEEDS to be in the official release. I still don't get why this isn't merged. This fixes an issue for LOTS of enthusiasts. The changes, as mentioned numerous times, are straight forward and non-intrusive.

vorner committed:
Disable sending crashdumps by default

We don't have consent from the user right now and they are not
encrypted. Shouldn't go to "production".

BFW-3468.

Derevin committed:
Remove uploading dumps to server

joachimklug committed:
fix: typo in internal server error

bkerler committed:
Fix missing virtualenv on running python build script

bkerler committed:
Fix german language issue with Filament Color layout and First setup screen

bkerler committed:
Fix buzzer issue by adding a wait for play time

Tomcus committed:
Remove default IP address for metrics & logs

bkerler committed:
gcode: M73 add missing get report functionality

danopernis committed:
Change phase stepping wizard into gcode

Commit:
Introduce HWLIMIT defintions for printers

BFW-5029

HelclM committed:
Basic click command support

- Accept CLICK_BUTTON event
- Parse the response
- Validates the dialog_id is still relevant
- Selects top FSM, validates the dialog_id and response for that FSM/phase and sends
  that Response

danopernis committed:
Fix focus stealing (not only) in DialogToolActionBox

Dialog stores the state of underlying screen upon construction
and loads it back upon destruction. Consistently. Like, always.

Without this, ToolBox::DialogToolActionBox would set focused item
which would prevent menu screen from serializing its focused item.

CZDanol-prusa committed:
Pause: unify loop_load_common & loop_load_mmu

BFW-5142

CZDanol-prusa committed:
Pause: MMU Load to nozzle - ask for color correct

BFW-5142

CZDanol-prusa committed:
Change text for MMU filament detected to something slightly more reasonable

michalrudolf committed:
Adjust retraction distance in UnloadInner (retract further back to ensure getting the filament out of the gear)

CZDanol-prusa committed:
Pause: remove unused (un)load_mode_t

CZDanol-prusa committed:
Pause: unify loop_load_mmu_change to _common

BFW-5043

CZDanol-prusa committed:
MMU: Ask color correct on M600/runout

BFW-5142 BFW-5043

danopernis committed:
Fix BSOD when aborting selftest while homing

CZDanol-prusa committed:
Unify Pause::loop_XX functions

There are four loop_load and four loop_unload functions
with almost identical implementation,
so we had duplicit code that was a PITA to maintain.

This commit unifies most of the functions,
there are still some less-similar smaller ones that can be unified
in the future.

Prerequisity for BFW-5043 and BFW-5142

kvmq committed:
Do not block HOST_USER_CLASS_ACTIVE callback with phost->class_mutex

kvmq committed:
More robust variant of usbh_power_cycle reset after an usb error

BFW-5178

vorner committed:
MMU: Count tool changes in the odometer

BFW-5138.

hejllukas committed:
IS: Fix the issue with a random printer stopping during printing.

vojtanvk committed:
Adding Bed Level Correction for MK3.5 and MINI

vorner committed:
MMU: Track failures in a leaky bucket

Not shown / used anywhere. But counts them, so we can use them later on.

BFW-5138.

danopernis committed:
Refactor frames in the phase stepping wizard

danopernis committed:
Introduce save_to_persistent_storage_without_enabling()

danopernis committed:
Remove obnoxious accelerometer logging

HelclM committed:
Rewrite state detection.

Newly it gurantees, that we only send dialogs from the top FSM. So we
don't have to do any gymnastics around determining for which FSM is the
click meant and just check dialog_id and if it matches it must be for
top FSM.

Also the state detection and dialog generation are handled seperatly, which
in theory makes it clearer. It introduces some similarities in the logic
in both places, but I'd say it is worth it, because otherwise it was
quiet messy and would be hard to guarantee to always send top FSM
dialog.

michalrudolf committed:
Fix bsod on mmu reflash

wavexx committed:
PS: Also preserve the step fraction with absolute origins

Re-integrate the missing mini-step before we recompute the absolute
origin, as previously done.

vorner committed:
connect: Allow SEND_STATE_INFO in errors

Allow them in redscreen/redscreen.

It's already sent proactively, so it's safe.

BFW-5207.

vorner committed:
MMU: Pop up warnings about extruder maintenance

BFW-5138.

vorner committed:
MMU: Menu item to reset the extruder maintenance

After the user does the cleaning of the extruder, this is where they can
reset the reminder.

BFW-5138.

danopernis committed:
Implement progress reporting in phase stepping calibration

vojtanvk committed:
Make MINI/3.5 stop digging into the bed

wavexx committed:
PS: Reduce duplication in reset_from_halt()

michalrudolf committed:
Fix XL Enclosure requested changes

Show Enclosure menu only if enclosure is active
Change string describing invalid temperature from "N/A" to "--"

wavexx committed:
PS: Preserve step fraction also when hard-stopping

Always try to preserve the step fraction, except when:

- reinitializing from scratch (no step fraction exists)
- after rehoming (position is assumed to be absolute)

This now includes the special case when a move is interrupted: if a move
involving only one axis is interrupted, we would previously loose the
fraction for all axes.

Since the generator state is ticking ahead, keep a new flag to keep
track of the axis state of all moves. Use this flag during reset to keep
the fraction when the axis hasn't actually moved, meaning we can perform
carry-over.

vorner committed:
MMU: Tests for the failure bucket

BFW-5138

danopernis committed:
Add header to phase stepping calibration wizard

danopernis committed:
Add hideous additional text to otherwise nice frame

CZDanol-prusa committed:
Include MMU2 FW in xFlash

BFW-3704

CZDanol-prusa committed:
MMU: FW Updater

BFW-3704

wavexx committed:
PS: Describe some fields

danopernis committed:
Rename reporter to hooks and make it into pure interface

danopernis committed:
Remove unused phase_stepping::last_calibration_result

danopernis committed:
Move phase_stepping::CalibrationResult to its only consumer

danopernis committed:
Remove some obsolete phase stepping abstractions

danopernis committed:
Enable aborting phase steping calibration wizard

danopernis committed:
Adjust phase stepping wizard final screen

HelclM committed:
Rename CLICK_BUTTON to DIALOG_ACTION

After discussion with connect we decided to rename the command from
CLICK_BUTTON to DIALOG_ACTION and the field "response" to "button". On
top of that just a couple changed in code names, so that it is
consistent.

danopernis committed:
M1977: Remove unused header

danopernis committed:
M1977: Change FSM only when there are valid data

danopernis committed:
M1977: Remove 'enabling' FSM state

HelclM committed:
Fix trasnfers not cleaning up

Transfers were not cleaned up ever, if in print preview when finished,
this was caused by wrongly detecting if the transfer is in progress. It
did not check, if the backup file was there, do transfer, where backup
file was not present (finished) were taken as in progress.

michalrudolf committed:
Add texts for coldpull dialog (for translations)
Update POT file

wavexx committed:
PS: Handle COREXY correctly when resetting

wavexx committed:
PS: Initialize current_distance directly with the stepper counters

Given:

- total_start_pos is now absolute and initialized from the counters, and
- step_generator_state.current_distance is the integer distance added so
  far to the counters already in physical units, and
- total_start_pos_msteps is actually current_distance + fraction

it follows we can avoid reconstructing current_distance from the move,
which incurs in extra rounding, but instead we can directly use the
stepper counters and leave total_start_pos_msteps unchanged when
resetting while preserving the fraction.

wavexx committed:
PS: Handle msteps.e uniformly when resetting

Always discard the fraction

wavexx committed:
PS: Avoid prefixing total_start_pos for consistency

... marginally makes the code more readable

wavexx committed:
PS: Also reset current_distance.e

since we reset msteps.e and completely ignore the absolute position for
the extruder.

HelclM committed:
Make Quick pause clickable in connect.

danopernis committed:
Hotfix bed level correction GUI after reset

CZDanol-prusa committed:
Hotfix: Show MI_HOTEND_TYPE on Mk3.5

BFW-5035

vorner committed:
MDNS: Tune timeouts vs delayed packets

BFW-5183.

CZDanol-prusa committed:
Add cmake error if MMU FW not present

BFW-3704

mody committed:
Finalize Cold Pull texts

Put the final wording to the proper places.

BFW-4858

michalrudolf committed:
Refactor Enclosure

CZDanol-prusa committed:
Hotfix: Wrong filaments button mode on non-MMU printers

BFW-3704

michalrudolf committed:
Update POT file

michalrudolf committed:
git subrepo commit (merge) lib/Prusa-Error-Codes

subrepo:
  subdir:   "lib/Prusa-Error-Codes"
  merged:   "2a53dd19f"
upstream:
  origin:   "git@github.com:prusa3d/Prusa-Error-Codes.git"
  branch:   "master"
  commit:   "7ff976755"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "110b9eb"

CZDanol-prusa committed:
Mk3.5: Callibrate selftest for HotendType::stock_with_sock

BFW-5238

CZDanol-prusa committed:
Increase ScrollBar width to 8 px

BFW-5050

CZDanol-prusa committed:
Fix freertos-gdb-plugin.py

BFW-5029

CZDanol-prusa committed:
Add stealth mode journal+marlin vars

BFW-5029

CZDanol-prusa committed:
Add stealth mode switch in the UI

Does nothing for now, funcitonality will be added later

BFW-5029

CZDanol-prusa committed:
Add stealth mode support for M73

BFW-5029

vorner committed:
MDNS: Make it active on only one interface

It is eating a bunch of different resources (we are discovering more and
more).

BFW-5183.

CZDanol-prusa committed:
Planner settings: prepare for applying machine limits

BFW-5029

vorner committed:
MDNS: Disable timeouts on iface removal

Apparently, mdns introduces bunch of timeouts, but didn't remove them on
removal of the interface. This resulted in two Bad Things:

* Hogging the timeouts.
* Having dangling pointers in them.

BFW-5183.

vorner committed:
MDNS: Further limits adjustments

* Adjust for only one interface with MDNS instead of both.
* Adjust for 5 more discovered timeouts per each interface :-(

BFW-5183.

CZDanol-prusa committed:
Planner: move max_jerk to settings

We're anticipating that we might want to limit it in stealth mode.

BFW-5029

CZDanol-prusa committed:
Add gcodes for stealth mode on/off

BFW-5029

CZDanol-prusa committed:
Planner: Apply HWLIMITs

BFW-5029

CZDanol-prusa committed:
Enable liveWatch in launch.json

vojtanvk committed:
Fix bad default change; Add SheetSupport option

HelclM committed:
Make dialogs take texts form error yaml.

Those texts are needed in the yaml for connect to show them, when we
send the codes. Take them from there even for dialogs, so that they are
not in two places nad out of sync.

rbuehlma committed:
Sync I2C code with marlin

rbuehlma committed:
Implement support for m260 and m261 (I2C read & write)

CZDanol-prusa committed:
Introduce gui_try_gcode_with_msg

CZDanol-prusa committed:
Use gui_try_gcode_with_msg in MI_STUCK_FILAMENT_DETECTION

vorner committed:
MMU: Change link for maintenance

prusa.io/mmu-maintenance -> prusa.io/mmu-care, as requested by Content

BFW-5232.

CZDanol-prusa committed:
Introduce MI_MMU_ISSUE_GCODE_SLOT

BFW-5152

CZDanol-prusa committed:
Use MI_MMU_ISSUE_GCODE_SLOT for MI_MMU_EJECT_FILAMENT

BFW-5152

CZDanol-prusa committed:
Use MI_MMU_ISSUE_GCODE_SLOT for MI_MMU_PRELOAD_SLOT

BFW-5152

CZDanol-prusa committed:
Use MI_MMU_ISSUE_GCODE_SLOT for MI_MMU_LOAD_TO_NOZZLE

BFW-5152

CZDanol-prusa committed:
Use MI_MMU_ISSUE_GCODE_SLOT for MI_MMU_CUT_FILAMENT

BFW-5152

CZDanol-prusa committed:
Use MI_MMU_ISSUE_GCODE_SLOT for MI_MMU_LOAD_TEST_FILAMENT

BFW-5152

CZDanol-prusa committed:
MMU: Rename Eject Filament -> Eject From MMU

BFW-5152

CZDanol-prusa committed:
Remove unused MI_CALIB_FSENSOR_MMU

BFW-5220

CZDanol-prusa committed:
SelftestFrameFSensor: Remove unused en_text_have_filament

BFW-5220

CZDanol-prusa committed:
SelftestFrameFSensor: Differentiate text for mmu_rework

BFW-5220

CZDanol-prusa committed:
Unify menu item style

Some menu items were not adhering to the casing standard.
Some also had colons at the end.

BFW-5266

CZDanol-prusa committed:
Enforce footer config when enabling MMU rework

BFW-5219

michalrudolf committed:
Edit frame_qr_layout for MINI

danopernis committed:
Rewrite FSM queues

Ditch smart queues and dumb queues.
Have the FSM state as the Marlin variable.
Let the clients do the thinking.

BFW-5215

HelclM committed:
Connect report state printing in regular MMU filament changes.

BFW-5270

CZDanol-prusa committed:
IWindowMenuItem: optimize move_focus to the same item

CZDanol-prusa committed:
Input Shaper menu: Disable editing during print

It screws up things - for example it resets weight compensation.

BFW-5251

CZDanol-prusa committed:
IS: M9200 gcode to reload settings from the store

BFW-5251

CZDanol-prusa committed:
IS menu: Use config_store instead of direct access

input_shaper module cannot be touched from the displayTask!
This was causing BSODs!

Instead, we adjust the settings in the config store
and issue a gcode that instructs the input_shaper
to reload settings.

BFW-5251

CZDanol-prusa committed:
Clear filament type when turning off MMU

BFW-5199

lukash committed:
Handle null pointers in crash dump save_message()

BFW-5264

HelclM committed:
git subrepo pull --force lib/Prusa-Error-Codes

subrepo:
  subdir:   "lib/Prusa-Error-Codes"
  merged:   "895dd5705"
upstream:
  origin:   "git@github.com:prusa3d/Prusa-Error-Codes.git"
  branch:   "master"
  commit:   "895dd5705"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "73a0129"

michalrudolf committed:
Fix dialog move z texts on MINI display

vojtanvk committed:
Fix MK3.5 MBL to be 7x7

CZDanol-prusa committed:
Fix WiInfo trying to translate EVERYTHING

Seems like there happened some hash conflict and we got translated something we didn't want to.

BFW-5246

CZDanol-prusa committed:
Refactor ScreenMenuVersionInfo

BFW-5246

danopernis committed:
Remove fsm_types_test and fsm_serializers_test

The extinction of old forms is the almost inevitable
consequence of the production of new forms.

BFW-5215

danopernis committed:
Remove test screens

BFW-5215

CZDanol-prusa committed:
Hotfix: snprintf warning in MI_MMU_ISSUE_GCODE_SLOT

BFW-5219

CZDanol-prusa committed:
Fix crash_dump_debug.py... Again.

BFW-5274

CZDanol-prusa committed:
Fix not checking for stream_gcode_start() in stream_thumbnail_start()

BFW-5274

CZDanol-prusa committed:
MMU: Update FW to 3.0.3

BFW-5263

Commit:
MMU:BFW-5144:Preload menu wording

Commit:
MMU:BFW-5144:Printer tools->MMU filament

Commit:
MMU:BFW-5144:MultioolFilChange->Filament Change

Commit:
MMU:BFW-5144:Tools Mapping->Filament mapping

Commit:
MMU:BFW-5144:Tool Filament N->Filament N

HelclM committed:
Fix locking in lwip

Forgot to lock for this one and disabling link from menu is something
not many people do.

vorner committed:
MMU maintenance texts

BFW-5231

CZDanol-prusa committed:
Fix PrusaPackGcodeReader::stream_t::hs_decoder being move unsafe

PrusaPackGcodeReader has a move constructor/operator.
stream_t::hs_decoder was unsafely moved around, potentially causing a memory corruption.

BFW-5274

CZDanol-prusa committed:
Make IGcodeReader constructors protected

BFW-5274

CZDanol-prusa committed:
Make IGcodeReader::file move-safe

BFW-5274

HelclM committed:
Small fixes for the new FSM

- don't commit changes, when it did not really changed, it was causing a
  change in the fsm generation and screwed up connect clicking

- send load unload data, not printing data to GetResponses in printer
  state

danopernis committed:
Do media_prefetch_init() inside the media_prefetch() task

danopernis committed:
Reduce heap allocations in media prefetch

CZDanol-prusa committed:
Another freertos-gdb-plugin fix.

CZDanol-prusa committed:
FIx FS/Estall lock missing in some loads

BFW-5279

CZDanol-prusa committed:
Use unique_file_ptr on more places

BFW-5274

CZDanol-prusa committed:
Fix memory leak in WindowPreviewThumbnail::unconditionalDraw

BFW-5274

michalrudolf committed:
Make hostname MI's visibility dev only

michalrudolf committed:
Show hostname MI on ili9488 display

CZDanol-prusa committed:
Fix one-click-print not triggering on USB reinsert

BFW-5281

michalrudolf committed:
Update translations 6.0.0

michalrudolf committed:
Add script for replacing unsupported characters in .PO files

vorner committed:
mmu: Adjust maintenance limits

BFW-5230.

michalrudolf committed:
Adjust MINI HW limits according to PrusaSlicer Profiles

michalrudolf committed:
Adjust MK3.5 HW limits according to PrusaSlicer Profiles

michalrudolf committed:
Adjust MK4 HW limits according to PrusaSlicer Profiles

Commit:
MMU:Register 0x82:Fail the next LoadToExtr

danopernis committed:
Admit using CMSIS in ChannelMutex

michalrudolf committed:
Remove Nextruder Maintanance menu item from MK3.5 FW

michalrudolf committed:
Extend MMU load to nozzle sequence on MK3.5

CZDanol-prusa committed:
Introduce M104.1 GCode

BFW-5282 BFW-4997

Commit:
Update translations: BFW-5288

michalrudolf committed:
Fix formatable error strings

michalrudolf committed:
Disable input shaper menu

CZDanol-prusa committed:
MMU: Fix reset sequence

The reset pin was floating.
Also some timing adjustments.

BFW-5286

CZDanol-prusa committed:
MMU: Add recovery from UART error

This error happens when the MMU is switching between the
bootloader and firmware - it gives up the control over the TX
and the Mk4 UART RX gets pulled down, which causes a frame error.

It gets pulled down because of the new pulldown R214 resistor on the xBuddy.

BFW-5286

danopernis committed:
Hotfix tool mapping screen after refactoring FSM.

As a special exception we do not close PrintPreview screen when the LoadUnload dialog
is requested. It would destroy the ToolsMappingBody while one of its methods is still
executing, leading to calling refresh_physical_tool_filament_labels() which in turn
jumped to undefined memory.

CZDanol-prusa committed:
MMU bootloader: Introduce result

BFW-5286

CZDanol-prusa committed:
MMU: MI_MMU_BOOTLOADER_RESULT

BFW-5286

bkerler committed:
Add mk3 compatibility to G90 command

mody committed:
Mark filament as unloaded after successful cold-pull

After successful cold-pull, there is no fillament loaded anymore
and we need to manually set the fillament as unloaded.

BFW-5304

CZDanol-prusa committed:
Stealth: auto fall back to std mode progress data

BFW-5305

CZDanol-prusa committed:
Use osDelay instead of HAL_Delay in usb_host

No need to pointlessly use up the processor

BFW-5213

danopernis committed:
Split gcode reader to separate files

CZDanol-prusa committed:
Do not call osDelay inside USBH_LL_DriverVBUS

Quite complex to explain why - please read the comments in the BFW-5213.
Basically, it gets called inside Tmr Svc during usb recovery,
but Tmr Svc cannot be blocking (by spec).

BFW-5213

CZDanol-prusa committed:
Fix missing MI_MMU_PRELOAD_ALL in ScreenMenuMMUPreloadToMMU

BFW-5310

michalrudolf committed:
Adjust loading to nozzle before print on mk3.5 with MMU + Fix absolute/relative positioning

CZDanol-prusa committed:
Decrease score of ClientFSM::Selftest

During selftest, a LoadUnload can be issued.
Before, the score of LoadUnload FSM was the same,
so the dialog was not shown.

BFW-5321

CZDanol-prusa committed:
Render ScreenMenuSTSCalibrations when dialog is shown

BFW-5321

CZDanol-prusa committed:
Fix wrong order in CSelftestPart_FSensor

The "Is filament unloaded" was showing right after user pressed on "unload",
without waiting for the unload dialog to finish.
It blinked a bit.

BFW-5321

CZDanol-prusa committed:
Get rid of selftest_snake::querying_user

Feels like it should be replaceable by checking if a dialog is open via GetFirstDialog()

BFW-5321

CZDanol-prusa committed:
Smaller header icon for USB

To fit the upcoming stealth icon :)

BFW-5313

CZDanol-prusa committed:
window_header_t: show/hide_bed_info -> set_show_bed_info

BFW-5313

CZDanol-prusa committed:
window_t::Shadow/Unshadow -> set_shadow

BFW-5313

CZDanol-prusa committed:
Refactor window_header_t::updateMedia

BFW-5313

CZDanol-prusa committed:
Improve window_header_t::updateAllRects

BFW-5313

CZDanol-prusa committed:
Add stealth mode icon to window_header

BFW-5313

CZDanol-prusa committed:
Get rid of the "beautiful rolling text" on home screen header

CZDanol-prusa committed:
Fix double mutex lock in media_print_resume

BFW-5329

CZDanol-prusa committed:
Add some asserts to GCodeReader

BFW-5329

CZDanol-prusa committed:
Fix uninitialized PrusaPackGcodeReader::ptr_stream_getc_decompressed

BFW-5329

CZDanol-prusa committed:
PlainGcodeReader::output_type -> IGcodeReader::stream_mode

BFW-5329

CZDanol-prusa committed:
GCodeReader:: add additional stream_mode checks

This should fix BSODs found by BFW-5329 & BFW-5274

CZDanol-prusa committed:
media_reset_position: Do not reset on reset failure

BFW-5329

CZDanol-prusa committed:
Improve PrusaPackGcodeReader::get_restore_block_for_offset

BFW-5329

michalrudolf committed:
Fix MK3.5 MMU after factory reset

Fix need for double turn on of the MMU (flip_mmu_rework turns on MMU)
Fix first print on fresh printer - properly set up logical/physical FS on MMU turn on/off

Commit:
MK35:FSensor:report synthetic filtered value

To make Sensor Info -> Filament Sensor raw value display 0 and 1.

BFW-5337

CZDanol-prusa committed:
media_print_reopen: Fix not updating restore info

BFW-5329

CZDanol-prusa committed:
Abstract out stream_restore_info_rec_t

Make stream restore info not PrusaPack only,
which saves us some ugly ifs and makes the interface cleaner.

BFW-5329

DRracer committed:
Hide Nextruder maintenance when no MMU rework

When the Nextruder hasn't been reworked into the MMU variant (i.e. Settings->HW->Nextruder is not MMU) -> do not show the "Nextruder maintenance" menu item.

BFW-5325

CZDanol-prusa committed:
fixup! Hide Nextruder maintenance when no MMU rework

Tomcus committed:
toolchange: slowdown parking to make tool offset calib more accurate

We noticed shifts in measured data. Recomend reading BFW-5276 for more
details, but in the end slowing down the parking speed helped a lot.

BFW-5276

CZDanol-prusa committed:
fixup! Get rid of selftest_snake::querying_user

BFW-5321

CZDanol-prusa committed:
Fix i2c::ChannelMutex

The original implementation:
- used dynamic allocation
- did not lock the mutex on the first instantiation
(get_handle was called before init_mutexes)
- relied on hi2c.Instance, which was null before I2C initialization
- had duplicit code (mutex inicialization)

The code is still ugly, but hopefully at least a bit better.

BFW-5274

CZDanol-prusa committed:
Improve Touchscreen_GT911::update_impl ISR check

Checking for the I2C handle was not enough.
For example, the handle gets unlocked on I2C reinit,
which kind of ruins things...

BFW-5274

CZDanol-prusa committed:
Fix BSOD on power panic resume

BFW-5344

Tomcus committed:
G425: Disable phase stepping when running Tool Offset Calibration

We shouldn't run any selftest with phasestepping enabled. To prevent
running G425 with phase stepping we are disabeling phasestepping
temporarily in the gcode it self instead of the selftest state machine.

This improves Tool Offset Calibration results when running said
calibration on already calibrated machine. The results are not perfect
yet. But this change removes the major inaccuracies in the measurements.

BFW-5276

CZDanol-prusa committed:
Hotfix: i2c::ChannelMutex

CZDanol-prusa committed:
Tool mapping: Do not require double touch

BFW-5346

CZDanol-prusa committed:
Header: larger 20x16 USB & stealth icons

BFW-5313

Tomcus committed:
PS: Make EnsureState movable

The valued that is moved from is invalidated to not reset values at the
end of its lifetime.

BFW-5322

Tomcus committed:
selftest: Disable phase stepping before running selftest

This should prevent noices when calibration X axis on XL and improve
Tool Offset Calibrations, but only when run as part of selftest, not as
gcode (will be done in separate commit/PR)

BFW-5322

Tomcus committed:
cleanup: Remove unused includes and remove not needed virtual methods

Small code cleanup.

BFW-5322

CZDanol-prusa committed:
Fix mistake in Backend::load_all

BFW-5335

CZDanol-prusa committed:
Config store backend: END_ITEM_SIZE_WITH_CRC, BANK_HEADER_SIZE_WITH_CRC

BFW-5335

Tomcus committed:
fix: Increase measurement task stack to run selftests in debug build

Tomcus committed:
calib Z: Disable PS when running test with gcode

BFW-5322

CZDanol-prusa committed:
journal::Backend: More reasonable free bank space calculation

BFW-5335

CZDanol-prusa committed:
journal::Backend: Introduce failing regression test

Fill will be in the next commit

BFW-5335

CZDanol-prusa committed:
Fix failing test journal::EEPROM::Regression BFW-3553

The problem was in write_end_item calling migrate_bank.
The function is called in store init, where migrate_bank would screw things.
In other cases, the free space for the end item is checked
in the parent calling functions.

BFW-3553

CZDanol-prusa committed:
Load: Fix Response::Stop not working in LoadPhases_t::check_filament_sensor_and_user_push__ask

BFW-5361

CZDanol-prusa committed:
Revert "Disable input shaper menu"

This reverts commit f605a28bea662f02ec8d5121f4f57cee8ada774b.

CZDanol-prusa committed:
Disable IS adjustments

BFW-5297

CZDanol-prusa committed:
Confirm print stop on serial printing

Copy pasted from standard printing screen

BFW-5371

CZDanol-prusa committed:
Fix screen_printing_data_t::w_etime_label not updating if value is N/A

BFW-5372

CZDanol-prusa committed:
Fix stealth mode showing 0% with old gcode

BFW-5305

CZDanol-prusa committed:
Translate FooterItemFinda values

BFW-5373

bkerler committed:
Change supported nozzle size from 0.2 to 1.2mm

danopernis committed:
Remove git remote check from build process

danopernis committed:
Only run tag creation and integration tests on private fork

danopernis committed:
Remove unused Dockerfile

danopernis committed:
Reduce pip dependency downloads for build stage

rbuehlma committed:
Remove unused code to fix CI build

3d-gussner committed:
Initial stale issuw workflow

Use same settings as we have in the 8-bit firmware see [here](https://github.com/prusa3d/Prusa-Firmware/blob/MK3/.github/workflows/stale.yml)

3d-gussner committed:
Update days

hawwwran committed:
cz-typo-fix

michalrudolf committed:
Fix M260 & M261 for MINI (wrong i2c interface)

thisiskeithb commented:
@CarlosGS: Would you be able to rebase/update this PR and resolve merge conflicts? I have a third-party Buddy board and would be keen to see if this fix still works.

CarlosGS committed:
Merge branch 'master' into patch-1
